### PR TITLE
hashsum: improve the error management to match GNU

### PR DESCRIPTION
**Pull request by sylvestre**
Should make tests/cksum/md5sum.pl pass

### CODE REVIEW

File: tests/by-util/test_hashsum.rs [hunk 1]
```diff
@@ -457,6 +457,24 @@ fn test_with_escape_filename_zero_text() {
     assert!(stdout.contains("a\nb"));
 }
 
+#[test]
+fn test_check_empty_line() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    at.write("f", "");
```
Comment (cakebaker) [inline tests/by-util/test_hashsum.rs:465]: I would simply use `at.touch("f");` here and in the other tests.

File: tests/by-util/test_hashsum.rs [hunk 2]
```diff
@@ -480,3 +498,115 @@ fn test_check_with_escape_filename() {
...
+    at.write("in.md5", "MD5(f)= d41d8cd98f00b204e9800998ecf8427f\n");
+    scene
+        .ccmd("md5sum")
+        .arg("--check")
+        .arg("--status")
+        .arg(at.subdir.join("in.md5"))
+        .fails()
+        .stderr_is("")
+        .stdout_is("");
```
Comment (cakebaker) [inline tests/by-util/test_hashsum.rs:563]:
```suggestion
        .no_output();
```

File: tests/by-util/test_hashsum.rs [hunk 3]
```diff
@@ -480,3 +498,115 @@ fn test_check_with_escape_filename() {
...
+    at.write("f", "");
+    at.write("in.md5", "d41d8cd98f00b204e9800998ecf8427f  f\n");
+    scene
+        .ccmd("md5sum")
+        .arg("--check")
+        .arg("--status")
+        .arg(at.subdir.join("in.md5"))
+        .fails()
+        .stderr_is("")
+        .stdout_is("");
```
Comment (cakebaker) [inline tests/by-util/test_hashsum.rs:580]:
```suggestion
        .no_output();
```

File: tests/by-util/test_hashsum.rs [hunk 4]
```diff
@@ -480,3 +498,115 @@ fn test_check_with_escape_filename() {
...
+}
+
+#[test]
+fn test_check_no_backslash_no_space() {
```
Comment (cakebaker) [inline tests/by-util/test_hashsum.rs:584]: I don't understand what this test is meant to test. The content of `in.md5` contains a space and while it's true that there is no backslash (except in `\n`) I don't see what the relevance is of there being no backslash.

File: src/uu/hashsum/src/hashsum.rs [hunk 5]
```diff
@@ -793,16 +809,51 @@ where
...
+            }
+            Ordering::Greater => {
+                show_warning_caps!("{} computed checksums did NOT match", failed_cksum)
+            }
+            Ordering::Less => {}
+        };
+
         match failed_open_file.cmp(&1) {
             Ordering::Equal => show_warning!("{} listed file could not be read", failed_open_file),
             Ordering::Greater => {
```
Comment (cakebaker) [inline src/uu/hashsum/src/hashsum.rs:859]: I think here and on line 860 `show_warning_caps!` should be used, too.

File: tests/by-util/test_hashsum.rs (hunk 4 above)
Comment (sylvestre) [reply tests/by-util/test_hashsum.rs:584]: it mirrors one of the test in GNU

File: tests/by-util/test_hashsum.rs [hunk 6]
```diff
@@ -244,7 +244,7 @@ fn test_check_file_not_found_warning() {
         .arg(at.subdir.join("testf.sha1"))
         .fails()
         .stdout_is("sha1sum: testf: No such file or directory\ntestf: FAILED open or read\n")
-        .stderr_is("sha1sum: warning: 1 listed file could not be read\n");
+        .stderr_is("sha1sum: WARNING: 1 listed file could not be read\n");
```
Comment (BenWiederhake) [inline tests/by-util/test_hashsum.rs:247]: The code potentially outputs a plural form. Is that part of our guarantee to the user?
- If yes, then it should be part of the tests.
- If no, then the code can be deleted/simplified.

**Review by BenWiederhake — APPROVED** [review]
Looks good. Two nits:
- I guess the seq commits should have been part of a separate PR, but whatever, let's merge it.
- The code handles singular/plural forms separately, but the tests don't check for it. Shouldn't that be tested?

File: tests/by-util/test_hashsum.rs (hunk 6 above)
Comment (sylvestre) [reply tests/by-util/test_hashsum.rs:247]: i will do it in a different PR, I have other changes pending

**Comment by sylvestre** [comment]
the seq change was a mistake, I removed it

**Pull request merged**